src/types/EmbeddedPaymentElement.tsx

```diff
@@ -0,0 +1,427 @@
+   * When using customerSessions, allow users to update their saved cards
+   */
+  updatePaymentMethodEnabled?: boolean;
+  /** The view can display payment methods like “Card” that, when tapped, open a sheet where customers enter their payment method details.
+   * The sheet has a button at the bottom. `formSheetAction` controls the action the button performs.
+   */
+  formSheetAction?: EmbeddedFormSheetAction;
+  /** Controls whether the view displays mandate text at the bottom for payment methods that require it. If set to `false`, your integration must display `PaymentOptionDisplayData.mandateText` to the customer near your “Buy” button to comply with regulations.
+   * Note: This doesn't affect mandates displayed in the form sheet.
+   */
+  embeddedViewDisplaysMandateText?: boolean;
+  /**
+   * Determines the behavior when a row  is selected. Defaults to `.default`.
+   */
+  rowSelectionBehavior?: RowSelectionBehavior;
+}
+
+// -----------------------------------------------------------------------------
+// Embedded API
+// -----------------------------------------------------------------------------
+
+class EmbeddedPaymentElement {
+  /**
+   * Call this when the intent configuration changes (e.g., amount or currency).
+   * Cancels any in-progress update. Ensures the correct payment methods are shown and fields are collected.
+   * If the selected payment option becomes invalid, it may be cleared.
+   * Returns the final result of the update; earlier in-flight updates will return `{ status: 'canceled' }`.
+   */
+  async update(
+    intentConfig: PaymentSheetTypes.IntentConfiguration
+  ): Promise<EmbeddedPaymentElementUpdateResult> {
+    const result =
+      await NativeStripeEmbedded.updateEmbeddedPaymentElement(intentConfig);
+    return result;
+  }
+
+  /**
+   * Confirm the payment or setup intent.
+   * Waits for any in-progress `update()` call to finish before proceeding.
+   * May present authentication flows (e.g., 3DS) if required.
+   * Requires the most recent `update()` call to have succeeded.
+   * Returns the final result: success, failure, or cancellation.
+   */
+  async confirm(): Promise<EmbeddedPaymentElementResult> {
+    const result = await NativeStripeEmbedded.confirmEmbeddedPaymentElement();
+    return result;
+  }
+
+  /** Clear the currently selected payment option (reset to null). */
+  clearPaymentOption(): void {
+    NativeStripeEmbedded.clearEmbeddedPaymentOption();
+  }
+}
+
+// -----------------------------------------------------------------------------
+// JS Factory: createEmbeddedPaymentElement
+// -----------------------------------------------------------------------------
+let confirmHandlerCallback: EmitterSubscription | null = null;
+let formSheetActionConfirmCallback: EmitterSubscription | null = null;
+let rowSelectionImmediateActionCallback: EmitterSubscription | null = null;
+
+async function createEmbeddedPaymentElement(
+  intentConfig: PaymentSheetTypes.IntentConfiguration,
```

Comment:
At least on Android we can't initialize with the intent config or the configuration. This is to support process death and activity recreation.

Reply:
Should be something we can handle on the bridging layer I'd assume, e.g. just hold onto the intentConfig and configuration until we can init.